mmv1/products/discoveryengine/SearchEngine.yaml

```diff
@@ -0,0 +1,153 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'SearchEngine'
+description: |
+  Vertex AI Search and Conversation can be used to create a search engine or a chat application by connecting it with a datastore
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Create a Search Engine': 'https://cloud.google.com/generative-ai-app-builder/docs/create-engine-es'
+  api: 'https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.collections.engines'
+
+base_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+self_link: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines?engineId={{engine_id}}'
+delete_url: 'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}'
+
+update_verb: :PATCH
+update_mask: true
+
+autogen_async: true
+
+async: !ruby/object:Api::OpAsync
+  actions: ['create', 'delete']
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+  result: !ruby/object:Api::OpAsync::Result
+    resource_inside_response: true
+
+import_format:
+  [
+    'projects/{{project}}/locations/{{location}}/collections/{{collection_id}}/engines/{{engine_id}}',
+  ]
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/discovery_engine_search_engine_hardcode_solution_type.go.erb
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "discoveryengine_searchengine_basic"
+    primary_resource_id: 'basic'
+    primary_resource_name:
+      'fmt.Sprintf("tf_test_search_engine%s", context["random_suffix"])'
+    vars:
+      engine_id: "example-engine-id"
+      data_store_id: "example-datastore-id"
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'engine_id'
+    description: |
+      Unique ID to use for Search Engine App.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'collection_id'
+    description: |
+      The collection ID.
+    required: true
+    immutable: true
+    url_param_only: true
+  - !ruby/object:Api::Type::String
+    name: 'location'
+    description: |
+      Location.
+    required: true
+    immutable: true
+    url_param_only: true
+
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    description: |
+      The unique full resource name of the search engine. Values are of the format
+      `projects/{project}/locations/{location}/collections/{collection_id}/engines/{engine_id}`.
+      This field must be a UTF-8 encoded string with a length limit of 1024
+      characters.
+    output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'industryVertical'
```

Comment:
We may need to set default value for this field since the default is fixed. Does the API return the default value for this field if it's not explicitly set? I don't see a test config without this field, so I'm not sure the API behavior. We may consider removing all optional fields from the basic example to ensure we have required fields set correctly and we handle fields that have default values returned from the API.

Reply:
SG. Let's add "default_value: :GENERIC". This is the default server behavior.

Reply:
I tested the provider in a TF directory without the industry_vertical field and it takes the default as generic. I will also add its default_value under properties

Reply:
Also for the Search and Chat Engine the industry_field is derived from datastore's industry_vertical so it has a dependency that's why is the example also I refer it to the Datastore instead of leaving it out. 
@shuyama1 Let me know your thoughts